presto-spi/src/main/java/io/prestosql/spi/type/Type.java

```diff
@@ -155,4 +158,36 @@
      * Compare the values in the specified block at the specified positions equal.
      */
     int compareTo(Block leftBlock, int leftPosition, Block rightBlock, int rightPosition);
+
+    /**
+     * Return the range of possible values for this type, if available.
+     *
+     * The type of the values must match {@link #getJavaType}
+     */
+    default Optional<Range> getRange()
+    {
+        return Optional.empty();
+    }
+
+    final class Range
```

Comment:
equals/hashCode are unnecessary, since we don't use them as map keys or compare ranges. It also doesn't make sense, since you can't compare ranges for different types anyway.

toString is not that useful because the min/max values are the native stack representation, which don't have a good rendering for things like Slice or Block.